src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -171,11 +170,29 @@ export class AddScenarioModal implements OnInit, OnDestroy {
 
     addError() {
         const control = <FormArray>this.newScenForm.get('regressionRegions.regressions.errors');
-        control.push(new FormControl(null, Validators.required));
+        control.push(this._fb.group({
+            id: new FormControl(null, Validators.required),
+            value: new FormControl(null, Validators.required)
+        }));
+    }  
+
+    fillError(id,index){
+        const errorList = <FormArray>this.newScenForm.get('regressionRegions.regressions.errors');
+        const error = errorList.at(index).value;
+        const selectedError = this.errors.find(x => x.id == id);
+        error.name = selectedError.name;
+        error.code = selectedError.code;
```

Comment:
So this is interesting: somewhere along the way the code and name are getting lost. I'm guessing it's because you didn't put code and name in the error form group (in the addError() function)
![image](https://user-images.githubusercontent.com/35497731/72624162-100e8800-390c-11ea-88c8-9623872c58b0.png)

What this means is that we only need the id and value of the error, not the name and code, so we can remove that whole fillError section here, as well as the call in the .html, which I'll also comment on